### ISSUE TITLE
Python 2.7.15 release notes are not available

### ISSUE THREAD

**Issue by mhsmith** [python/pythondotorg#1252]
Python 2.7.15 was released yesterday. However, the release notes link on the [download page](https://www.python.org/downloads/) doesn't go anywhere.

I see the 2.7 branch has changed the way they maintain their release notes file, so that's probably related.

**Comment by pradyunsg** [comment]
https://www.python.org/downloads/release/python-2715/

I was able to reach the above link after a single click on the web page linked in OP.

**Comment by mhsmith** [comment]
Yes, but that page still doesn't contain the full release notes. Compare it to the [Python 2.7.14 page](https://www.python.org/downloads/release/python-2714/), which contains a "Full changelog" link.

And the "Release notes" link on the top-level downloads page is still broken.

**Comment by pradyunsg** [comment]
Aha. I see what you mean. I misread. My bad.

Maybe @ned-deily has something to say here.

**Comment by ned-deily** [comment]
@benjaminp is the release manager for Python 2.7.x releases.

**Comment by pradyunsg** [comment]
Whoops. Sorry for the noise. :)

**Comment by brainwane** [comment]
@benjaminp Hi! I was trying to figure out whether 2.7.15 includes PEP 566 support (to resolve https://github.com/conda/conda-build/issues/2897#issuecomment-390487716 ) and couldn't find the release notes for 2.7.15. Ought I look at a combination of https://github.com/python/cpython/blob/2.7/Misc/NEWS.d/2.7.15rc1.rst and https://github.com/python/cpython/blob/2.7/Misc/NEWS.d/2.7.15.rst ?

Also, I see [someone on python-list](https://mail.python.org/pipermail/python-list/2018-May/732762.html) has also requested that we link to the 2.7.15 release notes.

**Comment by benjaminp** [comment]
Yes, the changes in 2.7.15 are the union of 2.7.15 and 2.7.15rc1 notes.

**Comment by dmiller-nmap** [comment]
Any plans to publish the release notes for 2.7.15?

**Comment by berkerpeksag** [comment]
The release notes for 2.7.15 is already published. I've just linked them from the download page: https://www.python.org/downloads/release/python-2715/ (so there is a "Full Changelog" link now)

I think switching to Blurb has introduced a regression here. I'd expect `2.7.15.rst` list all changes between 2.7.14 and 2.7.15 (including 2.7.15rc1), not only changes between 2.7.15.rc1 and 2.7.15 final.

**Comment by ned-deily** [comment]
Putting a link to just 2.7.t5.rst isn't a good solution as you note it's incomplete.  There were two relevant changes here.  A long ways back, the Python 3 documentation build was enhanced to produce a nicely formatted html version of the branch's changelog.  That enhancement and other changes to the overall docset format were not backported to 2.7.  Then, with the introduction of blurb to Python 2.7 and 3.x, there is no longer a single version-controlled file that contains the entire changelog;  in the past the 2.7.x release pages could just point to the repo copy of that file.  The Python 3.x doc builds were changed to use blurb to produce the html-formatted changelog but no similar change was made for 2.7 since blurb was not designed to produce the kind of plaintext changelog like the old 2.7 NEWS files.  So either the 2.7 docs could be enhanced to produce a similar html changelog, which is kind of a big deal since it affects any process that builds the 2.7 docs, or a plain text version of the changelog could be created possibly manually and stored somewhere where it could be served from python.org.

**Comment by berkerpeksag** [comment]
> Putting a link to just 2.7.t5.rst isn't a good solution as you note it's incomplete.

Agreed, I've tweaked it a bit:

![screenshot from 2018-11-09 01 27 13](https://user-images.githubusercontent.com/26338/48231296-a9fcab80-e3be-11e8-8648-9a91c83671e2.png)

(Feel free to suggest better wording.)

We can probably don't do anything, and link release notes manually (like I just did) until 2020 :)

**Comment by larryhastings** [comment]
For Python 3, building the docs runs "blurb merge", which generates what's essentially the old Misc/NEWS file.  This then gets rst2html-ified and finds its final destination as "changelog.html".  Then when we cut the release we link to the appropriate anchor on the release page, e.g.:

https://docs.python.org/3/whatsnew/changelog.html#python-3-7-1-final

Rather than... whatever it is you're proposing, why don't we also do that for Python 2?

**Comment by davidfraser** [comment]
The links on the [main release page](https://www.python.org/downloads/) to _Release Notes_ are still less than desirable:
* 2.7.15 just links back to the downloads page itself
* 2.7.16 just links to the final change to 2.7.16, not to the full changelog.

Additionally the [2.7.16 download page](https://www.python.org/downloads/release/python-2716/) just links to the final change to 2.7.16, not the 2.7.16rc1 changes too

**Comment by MANISH007700** [comment]
Go for the recent python version . I guess that's py 3.9(beta ) and py 3.8 stable !
have a good day :)
Link : 
https://www.python.org/downloads/
